uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerHost.java

```diff
@@ -292,16 +327,111 @@ private final String basicButtonHandler(@NonNull final String resourceButtonId,
             UiAutomatorUtils.handleButtonClick("android:id/button1");
         }
     }
-
-    @Override
-    public void setFlights(@Nullable final String flightsJson) {
-        Logger.i(TAG, "Set Flights..");
+    /**
+     * Gets all the accounts added
+     */
+    public List<String> getAllAccounts(boolean expectingMultipleAccounts) {
         launch();
-        // Make the set flights UI visible on screen
-        UiAutomatorUtils.obtainChildInScrollable("Set Flights");
-        // input flights string in flights input box
-        UiAutomatorUtils.handleInput("com.microsoft.identity.testuserapp:id/editTextFlights", flightsJson);
-        // Click Set Flights button
-        UiAutomatorUtils.handleButtonClick("com.microsoft.identity.testuserapp:id/setFlightsButton");
+
+        final String resourceButtonId = "com.microsoft.identity.testuserapp:id/buttonGetAccounts";
+        final String textId = "AccountName";
+        if (shouldHandleFirstRun) {
+            handleFirstRun(); // handle first run experience
+        }
+        UiAutomatorUtils.handleButtonClick(resourceButtonId);
+        // Look for the dialog box
+        final UiObject dialogBox = UiAutomatorUtils.obtainUiObjectWithResourceId(
+                "android:id/message"
+        );
+        Assert.assertTrue(dialogBox.exists());
+
+        // As of now we are testing only for 2 accounts.
+        List<String> accounts = new ArrayList<>();
+        final String accountName = getDialogBoxText(dialogBox, textId);
+        if (accountName != null) {
+            accounts.add(accountName);
+        }
+        if (expectingMultipleAccounts) {
+            final String accountName2 = getDialogBoxText(dialogBox, textId);
```

Comment:
This is interesting...this is calling the same method with same params i.e. `getDialogBoxText(dialogBox, textId);` but then how does it return a different account compared to the first invocation?

Reply:
It returns 2 accounts when we tap on the getAccounts button. which will show 2 dialog boxes one on top of the other. So when we do getDialogBoxText for the first time, we read the txt, get the first account and click on ok to dismiss the dialog. now we will still see the 2nd dialog box with a different account.

Reply:
Okay, so we're just looping through the dialogs here. I think the assumption around 2 accounts is okay for now thought we may need to refine it later.